azurerm/resource_arm_iothub.go

```diff
@@ -248,6 +248,51 @@ func resourceArmIotHub() *schema.Resource {
 				},
 			},
 
+			"fallback_route": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "$fallback",
+							ValidateFunc: validation.StringLenBetween(0, 64),
```

Comment:
Could we use a do a bit stronger of validation here:
```
				ValidateFunc: validation.StringMatch(
					regexp.MustCompile("^[-_.a-zA-Z0-9]{1,64$"),
					"Fallback Route Name name can only include alphanumeric characters, periods, underscores, hyphens, has a maximum length of 64 characters, and must be unique.",
				),
```

Reply:
Updated the validation for both FallbackRoute and Route.